controllers/profiles/following.js

```diff
@@ -8,14 +8,28 @@ const {QueryTypes} = require('sequelize');
 const config = require('../../databases/config/database')[env];
 
 module.exports = async (req, res) => {
-  const id = req.userId;
-  console.log(id);
+  const userId = req.userId;
+
   let sequelize;
   if (config.use_env_variable) {
     sequelize = new Sequelize(process.env[config.use_env_variable], config);
   } else {
     sequelize = new Sequelize(config.database, config.username, config.password, config);
   }
+
+  let user_topics = await sequelize.query(
+    `select tp.topic_id from topics as tp
+    left join user_topics as utp on tp.topic_id = utp.topic_id
+    where utp.user_id = :userId`,
+    {
+      type: QueryTypes.SELECT,
+      replacements: {
+        userId
+      }
+    }
+  );
+  let topicIds = user_topics.map((topic) => topic.topic_id);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The code looks good, but there's a potential performance issue. The `user_topics` query is executed every time this function is called. If the user's topics don't change frequently, consider caching the result to improve performance.

```diff
-  let user_topics = await sequelize.query(
-    `select tp.topic_id from topics as tp
-    left join user_topics as utp on tp.topic_id = utp.topic_id
-    where utp.user_id = :userId`,
-    {
-      type: QueryTypes.SELECT,
-      replacements: {
-        userId
-      }
-    }
-  );
-  let topicIds = user_topics.map((topic) => topic.topic_id);
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->